### PR TITLE
Feat(crm): Add documentation about look and feel of the CRM

### PR DESCRIPTION
**Pull request by arimet**

### CODE REVIEW

File: examples/crm/README.md [hunk 1]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+
+## Look and Feel
+
+This project offers a customizable CRM component. It leverages React context, allowing props to be accessed in any child component. By default, the values are sourced from the `crm.config.ts` file, but you can also override these props directly through the CRM component.
+
+```tsx
+// App.tsx
+import React from 'react';
+import CRM from './CRM';
+import { crmConfig } from './crm.config';
```
Comment (fzaninotto) [inline examples/crm/README.md:62]: not used in the example

File: examples/crm/README.md [hunk 2]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
 ### `npm run deploy`
 
 Deploy the build to GitHub gh-pages.
+
+## Look and Feel
+
+This project offers a customizable CRM component. It leverages React context, allowing props to be accessed in any child component. By default, the values are sourced from the `crm.config.ts` file, but you can also override these props directly through the CRM component.
```
Comment (fzaninotto) [inline examples/crm/README.md:56]:
```suggestion
You can customize the title, logo, theme, and domain of the CRM app by passing custom props to the `<CRM>` component:
```

File: examples/crm/README.md [hunk 3]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
 ### `npm run deploy`
 
 Deploy the build to GitHub gh-pages.
+
+## Look and Feel
```
Comment (fzaninotto) [inline examples/crm/README.md:54]:
```suggestion
## Customizing
```

File: examples/crm/README.md [hunk 4]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+        title="Custom CRM Title" 
+        logo="custom-logo.png" 
+    />
+);
+
+export default App;
+```
+You can also pass a custom `lightTheme` object to the CRM component to override the default React-admin theme. While React-admin supports `darkTheme`, you can add a theme switch button by including a ToggleThemeButton component in the Toolbar. For more information, refer to the [React-admin theme documentation](https://marmelab.com/react-admin/AppTheme.html).
+
+## Domain & Process
```
Comment (fzaninotto) [inline examples/crm/README.md:75]: I'd remove this section

File: examples/crm/README.md [hunk 5]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+
+In addition to the design, you can easily customize various aspects relevant to your business domain. The behavior is the same as described above. You can modify the following:
+
+| Props          | Description                                   | Type     |
+|----------------|-----------------------------------------------|----------|
+| companySectors | The industry sectors of the companies you add | string[] |
```
Comment (fzaninotto) [inline examples/crm/README.md:81]: show all props here

File: examples/crm/README.md [hunk 6]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+| taskTypes      | Types that defined a task    | string[] |
+
+
+```tsx
+import { CRM } from './CRM/CRM';
+import { crmConfig } from './CRM/crm.config';
+
+const App = () => (
+    <CRM
+        title={crmConfig.title}
```
Comment (fzaninotto) [inline examples/crm/README.md:94]: A good documentation shows example values for each prop

File: examples/crm/README.md [hunk 7]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+        dealCategories={crmConfig.dealCategories}
+        noteStatuses={crmConfig.noteStatuses}
+        taskTypes={crmConfig.taskTypes}
+    />
+);
+
+export default App;
+```
+
+## Add Sales
```
Comment (fzaninotto) [inline examples/crm/README.md:107]:
```suggestion
## Adding Sales
```

File: examples/crm/README.md [hunk 8]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+
+To add a new sale to the CRM, you need to use an administrator account. By default, the first account created has this role. If you are starting fresh, a sign-up page will prompt you to create this admin account.
+
+When logged in as an admin, an 'Account Manager' tab will be available. From this page, you can create sales and transfer the administrator role.
```
Comment (fzaninotto) [inline examples/crm/README.md:111]: Please add a screenshot

File: examples/crm/README.md [hunk 9]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+
+## Customize the Homepage
```
Comment (fzaninotto) [inline examples/crm/README.md:113]:
```suggestion
## Customizing the Homepage
```

File: examples/crm/README.md [hunk 10]
```diff
@@ -50,3 +50,66 @@ Your app is ready to be deployed!
...
+
+The first page of the application is managed by the `Dashboard.tsx` component. You can customize it by updating this file.
```
Comment (fzaninotto) [inline examples/crm/README.md:115]: PLease show an example

File: examples/crm/src/CRM/CRM.tsx
Comment (fzaninotto) [inline examples/crm/src/CRM/CRM.tsx:1]: I'd call the directory `root` instead of `CRM`

File: examples/crm/src/CRM/CRMContext.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    logo?: string;
+}
+
+// Define types for the CRMProvider props
+interface CRMProviderProps extends CRMContextValue {
+    children: ReactNode;
+}
+
+// Create context with default value
+export const CRMContext = createContext<CRMContextValue>({
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRMContext.tsx:21]: I'd call it `ConfigurationContext` instead

File: examples/crm/src/CRM/CRMContext.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
+import { createContext, ReactNode, useContext } from 'react';
+import { crmConfig } from './crm.config';
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRMContext.tsx:2]: I'd call it `defaultConfiguration`.

File: examples/crm/src/CRM/CRMContext.tsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    companySectors: crmConfig.companySectors,
+    dealCategories: crmConfig.dealCategories,
+    dealStages: crmConfig.dealStages,
+    noteStatuses: crmConfig.noteStatuses,
+    taskTypes: crmConfig.taskTypes,
+    title: crmConfig.title,
+    logo: crmConfig.logo,
+});
+
+export const CRMProvider = ({
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRMContext.tsx:31]: `ConfigurationProvider`?

File: examples/crm/src/CRM/CRMContext.tsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+// Define types for the context value
+export interface CRMContextValue {
+    companySectors?: string[];
+    dealCategories?: string[];
+    dealStages?: { value: string; label: string }[];
+    noteStatuses?: { value: string; label: string; color: string }[];
+    taskTypes?: string[];
+    title?: string;
+    logo?: string;
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRMContext.tsx:12]: the values of each key must be defined. You need to remove the `?`s.

File: examples/crm/src/CRM/CRMContext.tsx [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+// Define types for the CRMProvider props
+interface CRMProviderProps extends CRMContextValue {
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRMContext.tsx:16]: You should export this interface, too.

File: examples/crm/src/CRM/CRMContext.tsx [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+    noteStatuses: crmConfig.noteStatuses,
+    taskTypes: crmConfig.taskTypes,
+    title: crmConfig.title,
+    logo: crmConfig.logo,
+});
+
+export const CRMProvider = ({
+    children,
+    companySectors = crmConfig.companySectors,
+    dealCategories,
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRMContext.tsx:34]: why no default value?

File: examples/crm/src/CRM/crm.config.ts [hunk 17]
```diff
@@ -0,0 +1,63 @@
+const logo =
```
Comment (fzaninotto) [inline examples/crm/src/CRM/crm.config.ts:1]: you should export each individual config key instead of one parent

File: examples/crm/src/CRM/CRM.tsx [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+    lightTheme?: RaThemeOptions;
+    title?: CRMContextValue['title'];
+    companySectors?: CRMContextValue['companySectors'];
+    dealStages?: CRMContextValue['dealStages'];
+    dealCategories?: CRMContextValue['dealCategories'];
+    noteStatuses?: CRMContextValue['noteStatuses'];
+    taskTypes?: CRMContextValue['taskTypes'];
+}
+
+export const CRM = ({
```
Comment (fzaninotto) [inline examples/crm/src/CRM/CRM.tsx:37]: Please add a JSDoc with basic usage

File: examples/crm/src/companies/CompanyShow.tsx [hunk 19]
```diff
@@ -275,7 +276,12 @@ const DealsIterator = () => {
                             secondary={
                                 <>
                                     {/* @ts-ignore */}
```
Comment (fzaninotto) [inline examples/crm/src/companies/CompanyShow.tsx:278]: You probably don't need that anymore

File: examples/crm/src/companies/CompanyShow.tsx [hunk 20]
```diff
@@ -275,7 +276,12 @@ const DealsIterator = () => {
...
-                                    {stageNames[deal.stage]},{' '}
+                                    {
+                                        dealStages?.find(
```
Comment (fzaninotto) [inline examples/crm/src/companies/CompanyShow.tsx:280]: I think you've repeated this at least once, so it'd be a good idea to extract it to a function in the `deal` domain

File: examples/crm/src/contacts/ContactListFilter.tsx [hunk 21]
```diff
@@ -73,38 +75,19 @@ export const ContactListFilter = () => {
...
-                />
-                <FilterListItem
-                    label={
-                        <>
-                            In contract <Status status="in-contract" />
-                        </>
-                    }
-                    value={{ status: 'in-contract' }}
-                />
+                {noteStatuses &&
```
Comment (fzaninotto) [inline examples/crm/src/contacts/ContactListFilter.tsx:78]: `noteStatuses` must be defined, the error is in your type

File: examples/crm/src/dashboard/DealsPipeline.tsx [hunk 22]
```diff
@@ -30,11 +31,11 @@ export const DealsPipeline = () => {
             return;
         }
         const deals: Deal[] = [];
-        stages
-            .filter(stage => stage !== 'won')
+        dealStages
+            ?.filter(stage => stage.value !== 'won')
```
Comment (fzaninotto) [inline examples/crm/src/dashboard/DealsPipeline.tsx:35]: `?` should not be necessary. Also, the `won` value should be customizable. Something like `dealsPipelineStatuses`

File: examples/crm/src/dataGenerator/contactNotes.ts [hunk 23]
```diff
@@ -16,11 +14,10 @@ export const generateContactNotes = (db: Db): ContactNote[] => {
         return {
             id,
             contact_id: contact.id,
-            type: random.arrayElement(type),
```
Comment (fzaninotto) [inline examples/crm/src/dataGenerator/contactNotes.ts:19]: why no more type?

File: examples/crm/src/dataGenerator/contacts.ts [hunk 24]
```diff
@@ -10,9 +10,9 @@ import {
 import { randomDate, weightedBoolean } from './utils';
 import { Db } from './types';
 import { Contact } from '../types';
+import { crmConfig } from '../CRM/crm.config';
 
 const genders = ['male', 'female', 'nonbinary'];
```
Comment (fzaninotto) [inline examples/crm/src/dataGenerator/contacts.ts:15]: should be configurable

File: examples/crm/src/dataGenerator/dealNotes.ts [hunk 25]
```diff
@@ -3,16 +3,13 @@ import { random, lorem } from 'faker/locale/en_US';
...
-const type = ['Email', 'Call', 'Call', 'Call', 'Call', 'Meeting', 'Reminder'];
-
 export const generateDealNotes = (db: Db) => {
     return Array.from(Array(300).keys()).map(id => {
         const deal = random.arrayElement(db.deals);
         deal.nb_notes++;
         return {
             id,
             deal_id: deal.id,
-            type: random.arrayElement(type),
```
Comment (fzaninotto) [inline examples/crm/src/dataGenerator/dealNotes.ts:15]: Why did you remove this ?

File: examples/crm/src/dataGenerator/deals.ts [hunk 26]
```diff
@@ -46,8 +30,8 @@ export const generateDeals = (db: Db): Deal[] => {
             name: lowercaseName[0].toUpperCase() + lowercaseName.slice(1),
             company_id: company.id,
             contact_ids: contacts.map(contact => contact.id),
-            type: random.arrayElement(type),
```
Comment (fzaninotto) [inline examples/crm/src/dataGenerator/deals.ts:49]: Why did you remove the type?

File: examples/crm/src/dataGenerator/tasks.ts [hunk 27]
```diff
@@ -3,34 +3,7 @@ import { lorem, random } from 'faker/locale/en_US';
 import { Task } from '../types';
 import { Db } from './types';
 import { randomDate } from './utils';
-
-export const type: Task['type'][] = [
```
Comment (fzaninotto) [inline examples/crm/src/dataGenerator/tasks.ts:7]: Nope, this should stay in the data generator. It controls the likelihood of each option.

File: examples/crm/src/tasks/AddTask.tsx [hunk 28]
```diff
@@ -84,7 +85,7 @@ export const AddTask = () => {
                                 <SelectInput
                                     source="type"
                                     validate={required()}
-                                    choices={taskTypes.map(type => ({
+                                    choices={taskTypes?.map(type => ({
```
Comment (fzaninotto) [inline examples/crm/src/tasks/AddTask.tsx:88]: Should always be defined

File: examples/crm/src/tasks/TaskEdit.tsx [hunk 29]
```diff
@@ -66,7 +67,7 @@ export const TaskEdit = ({
                             <SelectInput
                                 source="type"
                                 validate={required()}
-                                choices={taskTypes.map(type => ({
+                                choices={taskTypes?.map(type => ({
```
Comment (fzaninotto) [inline examples/crm/src/tasks/TaskEdit.tsx:70]: should always be defined

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: examples/crm/src/dataGenerator/deals.ts (hunk 26 above)
Comment (arimet) [reply examples/crm/src/dataGenerator/deals.ts:49]: I didn't delete it, I just renamed it category, as this term is used on the user side and in the specifications.

File: examples/crm/src/dataGenerator/dealNotes.ts (hunk 25 above)
Comment (arimet) [reply examples/crm/src/dataGenerator/dealNotes.ts:15]: None of the note components display or use this property. Or maybe I've missed it

File: examples/crm/src/dataGenerator/contactNotes.ts (hunk 23 above)
Comment (arimet) [reply examples/crm/src/dataGenerator/contactNotes.ts:19]: None of the note components display or use this property. Or maybe I've missed it

File: examples/crm/src/dataGenerator/tasks.ts [hunk 30]
```diff
@@ -30,6 +33,7 @@ export const type: Task['type'][] = [
     'Thank you',
     'Ship',
     'None',
+    'toto',
```
Comment (jonathan-marmelab) [inline examples/crm/src/dataGenerator/tasks.ts:36]: Is this a debug test ?

File: examples/crm/src/notes/StatusSelector.tsx [hunk 31]
```diff
@@ -2,31 +2,28 @@ import * as React from 'react';
...
+            onChange={(event: React.ChangeEvent<{ value: unknown }>) => {
+                setStatus(event.target.value);
+            }}
+            variant="filled"
+            label={false}
+            margin="none"
+            size="small"
+            sx={sx}
+        >
+            {noteStatuses?.map(status => (
```
Comment (jonathan-marmelab) [inline examples/crm/src/notes/StatusSelector.tsx:22]:
```suggestion
            {noteStatuses.map(status => (
```

File: examples/crm/src/root/CRM.tsx [hunk 32]
```diff
@@ -0,0 +1,158 @@
...
+    defaultDealPipelineStatuses,
+    defaultDealStages,
+    defaultLogo,
+    defaultNoteStatuses,
+    defaultTaskTypes,
+    defaultTitle,
+} from './defaultConfiguration';
+
+// Define the interface for the CRM component props
+interface CRMProps {
```
Comment (jonathan-marmelab) [inline examples/crm/src/root/CRM.tsx:39]: This can be simplified to : 

```ts
type CRMProps = {
    lightTheme?: RaThemeOptions;
    darkTheme?: RaThemeOptions;
} & Partial<ConfigurationContextValue>;
```

File: examples/crm/src/deals/DealListContent.tsx [hunk 33]
```diff
@@ -71,11 +73,11 @@ export const DealListContent = () => {
     return (
         <DragDropContext onDragEnd={onDragEnd}>
             <Box display="flex">
-                {stages.map(stage => (
+                {dealStages?.map(stage => (
```
Comment (jonathan-marmelab) [inline examples/crm/src/deals/DealListContent.tsx:76]:
```suggestion
                {dealStages.map(stage => (
```

File: examples/crm/src/deals/DealList.tsx [hunk 34]
```diff
@@ -50,6 +50,19 @@ const DealLayout = () => {
...
 
+    const { dealCategories } = useConfigurationContext();
+
+    const dealFilters = [
+        <SearchInput source="q" alwaysOn />,
+        <ReferenceInput source="company_id" reference="companies" />,
+        <SelectInput
+            source="type"
+            label="Category"
+            choices={dealCategories?.map(type => ({ id: type, name: type }))}
```
Comment (jonathan-marmelab) [inline examples/crm/src/deals/DealList.tsx:61]:
```suggestion
            choices={dealCategories.map(type => ({ id: type, name: type }))}
```

File: examples/crm/src/deals/DealForm.tsx [hunk 35]
```diff
@@ -82,11 +82,21 @@ export const DealForm = () => {
...
+                    name: stage.label,
+                }))}
                 validate={validateRequired}
                 defaultValue="opportunity"
             />
-            <SelectInput source="type" label="Category" choices={typeChoices} />
+            <SelectInput
+                source="category"
+                label="Category"
+                choices={dealCategories?.map(type => ({
```
Comment (jonathan-marmelab) [inline examples/crm/src/deals/DealForm.tsx:95]:
```suggestion
                choices={dealCategories.map(type => ({
```

File: examples/crm/src/deals/DealForm.tsx [hunk 36]
```diff
@@ -82,11 +82,21 @@ export const DealForm = () => {
 
             <SelectInput
                 source="stage"
-                choices={stageChoices}
+                choices={dealStages?.map(stage => ({
```
Comment (jonathan-marmelab) [inline examples/crm/src/deals/DealForm.tsx:85]:
```suggestion
                choices={dealStages.map(stage => ({
```

File: examples/crm/src/dashboard/DealsPipeline.tsx [hunk 37]
```diff
@@ -30,11 +32,11 @@ export const DealsPipeline = () => {
             return;
         }
         const deals: Deal[] = [];
-        stages
-            .filter(stage => stage !== 'won')
+        dealStages
+            ?.filter(stage => !dealPipelineStatuses.includes(stage.value))
```
Comment (jonathan-marmelab) [inline examples/crm/src/dashboard/DealsPipeline.tsx:36]:
```suggestion
            .filter(stage => !dealPipelineStatuses.includes(stage.value))
```

File: examples/crm/src/companies/CompanyListFilter.tsx [hunk 38]
```diff
@@ -31,7 +36,7 @@ export const CompanyListFilter = () => {
             </FilterList>
 
             <FilterList label="Sector" icon={<LocalShippingIcon />}>
-                {sectors.map(sector => (
+                {sectors?.map(sector => (
```
Comment (jonathan-marmelab) [inline examples/crm/src/companies/CompanyListFilter.tsx:39]:
```suggestion
                {sectors.map(sector => (
```

File: examples/crm/src/companies/CompanyListFilter.tsx [hunk 39]
```diff
@@ -12,10 +12,15 @@ import LocalShippingIcon from '@mui/icons-material/LocalShipping';
 import SupervisorAccountIcon from '@mui/icons-material/SupervisorAccount';
 
 import { sizes } from './sizes';
-import { sectors } from './sectors';
+import { useConfigurationContext } from '../root/ConfigurationContext';
 
 export const CompanyListFilter = () => {
     const { identity } = useGetIdentity();
+    const { companySectors } = useConfigurationContext();
+    const sectors = companySectors?.map(sector => ({
```
Comment (jonathan-marmelab) [inline examples/crm/src/companies/CompanyListFilter.tsx:20]:
```suggestion
    const sectors = companySectors.map(sector => ({
```

File: examples/crm/src/companies/CompanyForm.tsx [hunk 40]
```diff
@@ -12,78 +12,84 @@ import {
...
-        <Divider sx={{ mb: 2, width: '100%' }} />
+export const CompanyForm = () => {
+    const { companySectors } = useConfigurationContext();
+    return (
+        <>
+            <TextInput source="name" validate={required()} />
+            <Stack direction="row">
+                <SelectInput
+                    source="sector"
+                    choices={companySectors?.map(sector => ({
```
Comment (jonathan-marmelab) [inline examples/crm/src/companies/CompanyForm.tsx:26]:
```suggestion
                    choices={companySectors.map(sector => ({
```

File: examples/crm/src/companies/CompanyCard.tsx [hunk 41]
```diff
@@ -51,7 +52,10 @@ export const CompanyCard = (props: { record?: Company }) => {
                         <SelectField
                             color="textSecondary"
                             source="sector"
-                            choices={sectors}
+                            choices={companySectors?.map(sector => ({
```
Comment (jonathan-marmelab) [inline examples/crm/src/companies/CompanyCard.tsx:55]:
```suggestion
                            choices={companySectors.map(sector => ({
```

**Review by jonathan-marmelab — CHANGES_REQUESTED** [review]

**Review by jonathan-marmelab — APPROVED** [review]
